R/zchunk_LA124.heat.R

```diff
@@ -69,117 +77,121 @@ module_energy_LA124.heat <- function(command, ...) {
     L124.heatoutratio_R_elec_F_tech_Yh %>%
       left_join(all_years_0) %>%
-      mutate(sum=if_else(is.na(sum), 2, sum)) %>% #2 was a random, nonzero choice
-      filter(sum!=0)  %>%
+      #2 is a random, non-zero option for temporary use
+      mutate(sum = if_else(is.na(sum), 2, sum)) %>%
+      filter(sum != 0)  %>%
       select(-sum) -> L124.heatoutratio_R_elec_F_tech_Yh
 
     # This is complicated. Some historical regions / years have 0 district heat output (all fuels). This is fine unless there is heat output from
     # IEA's MAINCHP flow, which we model in the electric sector with a secondary output of heat to the district heat sector. In GCAM this will
     # cause solution failure as the district heat supply will be 0. In order to avoid this outcome we need a nominal amount of heat produced
     # in the zero years. The code below sets an exogenous year as the one to use to extract fuel shares for each region. In all 0 region / years,
     # output is equal to the output in the fuel-share year times 1e-3 to avoid throwing off the energy balances.
-    fuel_share_year <- "2010"
+
+    fuel_share_year <- 2010
     L124.in_EJ_R_heat_F_Yh %>%
       group_by(sector, GCAM_region_ID, year) %>%
-      summarise(value=sum(value)) %>%
-      mutate(value=if_else(value==0, 1e-3, value)) %>%
-      mutate(value=if_else(value!=1e-3, 0, value)) -> L124.mult_R_heat_Yh
+      summarise(value = sum(value)) %>%
+      # Creating an intermediary table - for years with 0 heat output, set value to 1e-3, for all others, set value to 0
+      mutate(value = if_else(value == 0, 1e-3, 0)) -> L124.mult_R_heat_Yh
 
     L124.out_EJ_R_heatfromelec_F_Yh %>%
       group_by(sector, GCAM_region_ID, year) %>%
-      summarise(elec_heat=sum(value)) %>%
-      mutate(elec_heat=if_else(elec_heat!=0, 1, elec_heat))-> L124.mult_R_heatfromelec_Yh
+      summarise(elec_heat = sum(value)) %>%
+      # Creating an intermediary table - for years with nonzero heat from CHP, set value to 1, for 0 values, leave as 0
+      mutate(elec_heat = if_else(elec_heat != 0, 1, elec_heat))-> L124.mult_R_heatfromelec_Yh
 
+    # If heat output is 0 and heat from CHP is nonzero, multiplier will be nonzero (1e-3)
+    # All other cases, multiplier is 0
     L124.mult_R_heat_Yh %>%
       left_join(L124.mult_R_heatfromelec_Yh %>%
                   rename(temp=sector),
-                  by=c("year", "GCAM_region_ID")) %>%
-      mutate(value=value*elec_heat) %>%
+                  by = c("year", "GCAM_region_ID")) %>%
+      mutate(value = value * elec_heat) %>%
       select(-temp, -elec_heat) %>%
-      mutate(value=if_else(is.na(value), 0, value))-> L124.mult_R_heat_Yh
+      mutate(value = if_else(is.na(value), 0, value))-> L124.mult_R_heat_Yh
 
     L124.in_EJ_R_heat_F_Yh %>%
-      filter(year==fuel_share_year) %>%
+      filter(year == fuel_share_year) %>%
       select(-year) %>%
-      rename(fsy_val=value)-> fsy_in
+      rename(fsy_val = value)-> fsy_in
```

Comment:
related comment above at line 138. fsy_val could be hist_value or norm_value, fsy_in could be hist_in or norm_in.